File: factors.c
#include "factors.h"

/**
 * main - main function
 *
 * Return: void
 */
int main(int argc, char *argv[])
{
	FILE *file;
	size_t counter;
	ssize_t line;
	char *buffer = NULL;


	if (argc != 2)
	{
		fprintf(stderr, "Usage: factors <filename>\n");
		exit(EXIT_FAILURE);
	}

	file = fopen(argv[1], "r");
	if (file == NULL)
	{
		fprintf(stderr, "Error: can't open file %s\n", argv[1]);
		exit(EXIT_FAILURE);
	}

	while ((line = getline(&buffer, &counter, file)) != -1)
	{
		factorize(buffer);
	}

	return (0);
}

File: factors.py
import sys

def factorize(number):
    for i in range(2, number):
        if number % i == 0:
            return i, number // i
    return None

if len(sys.argv) != 2:
    print("Usage: factors <file>")
    sys.exit(1)

input_file = sys.argv[1]

try:
    with open(input_file, 'r') as file:
        for line in file:
            number = int(line)
            factors = factorize(number)
            if factors:
                p, q = factors
                print(f"{number}={p}*{q}")

except FileNotFoundError:
    print(f"Error: File {input_file} not found.")

except ValueError:
    print(f"Error: Invalid input in file {input_file}.")

except Exception as e:
    print(f"An error occurred: {str(e)}")

